### PR TITLE
Adding managerId field to /package/install and /package/uninstall calls

### PR DESCRIPTION
**Pull request by kvish**
adding manager Id field to support custom manager changes for the k8s team. 

Addresses DCOS-40422

### CODE REVIEW

File: python/lib/dcos/dcos/packagemanager.py [hunk 1]
```diff
@@ -98,23 +98,26 @@ def enabled(self):
...
         :type pkg: CosmosPackageVersion
         :param options: user supplied package parameters
         :type options: dict
         :rtype: None
+        :param manager_id: the custom manager to forward this request to
+        :type manager_id: str
         """
 
-        params = {"packageName": pkg.name(), "packageVersion": pkg.version()}
+        params = {"packageName": pkg.name(), "packageVersion": pkg.version()
```
Comment (takirala) [inline python/lib/dcos/dcos/packagemanager.py]: Is this missing a comma ?!

File: python/lib/dcoscli/dcoscli/package/main.py [hunk 2]
```diff
@@ -73,7 +73,7 @@ def _cmds():
         cmds.Command(
             hierarchy=['package', 'install'],
             arg_keys=['<package-name>', '--package-version', '--options',
-                      '--cli', '--global', '--app', '--yes'],
+                      '--cli', '--global', '--app', '--yes', '--manager-id'],
```
Comment (takirala) [inline python/lib/dcoscli/dcoscli/package/main.py]: I think we agreed to hide the `managerId` field from the user help menu. Is that what's being done here?

File: python/lib/dcoscli/dcoscli/package/main.py [hunk 3]
```diff
@@ -89,7 +89,7 @@ def _cmds():
         cmds.Command(
             hierarchy=['package', 'uninstall'],
             arg_keys=['<package-name>', '--all', '--app-id', '--cli', '--app',
-                      '--yes'],
+                      '--yes', '--manager-id'],
```
Comment (takirala) [inline python/lib/dcoscli/dcoscli/package/main.py]: (nit) I think `--manage-id` can precede `--yes`

File: python/lib/dcoscli/dcoscli/package/main.py [hunk 4]
```diff
@@ -317,6 +317,7 @@ def _install(package_name, package_version, options_path, cli,
     :type yes: bool
     :returns: process status
     :rtype: int
+    :manager_id: str defines the custom manager to forward this operation to
```
Comment (takirala) [inline python/lib/dcoscli/dcoscli/package/main.py]: (applies to other parts of the file too). All the pydocs for method params should come before `:rtype`

**Comment by bamarni** [comment]
@kvish : there are some linting issues 

```
dcos/package.py:57:30: W291 trailing whitespace
dcos/package.py:58:28: W291 trailing whitespace
dcos/package.py:59:24: W291 trailing whitespace
dcos/package.py:60:5: E125 continuation line with same indent as next logical line
dcos/package.py:61:1: W293 blank line contains whitespace
```

**Comment by bamarni** [comment]
You can check them with : 
```
 cd python/lib/dcos
make env
source env/bin/activate
tox -e py35-syntax
```

File: python/lib/dcoscli/dcoscli/package/main.py (hunk 2 above)
Comment (bamarni) [reply python/lib/dcoscli/dcoscli/package/main.py]: I don't think it's easily doable, the library it's using (docopt) relies on the help menu to parse available commands and flags.

Though I'm surprised that this change works if you didn't change https://github.com/dcos/dcos-core-cli/blob/master/python/lib/dcoscli/dcoscli/data/help/package.txt.

**Comment by takirala** [comment]
For future reference, this PR is closed as it was not trivial to hide `managerId` field. If we need to do this in future https://github.com/docopt/docopt/issues/232 has some interesting discussion.

**Pull request closed without merging**